libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookHelper.cs

```diff
@@ -75,7 +75,7 @@ public static Activity ProcessSingleMessage(FacebookMessage message)
                 return null;
             }
 
-            if (message.Sender == null)
+            if (message.Sender == null && message.Optin != null && message.Optin.UserRef != null)
```

Comment:
I don't know what this is but I assume it's correct. Anyway, could be changed to
`message.Sender == null && message.Optin?.UserRef != null` I think